datumaro/plugins/sampler/sampler.py

```diff
@@ -0,0 +1,212 @@
+        algorithm,
+        subset_name,
+        sampled_name,
+        unsampled_name,
+        sampling_method,
+        num_sample,
+        output_file,
+    ):
+        """
+        Parameters
+        ----------
+        extractor : Extractor, Dataset
+        algorithm : str
+            Specifying the algorithm to calculate the uncertainty
+            for sample selection. default: 'entropy'
+        subset_name : str
+            The name of the subset to which you want to select a sample.
+        sample_name : str
+            Subset name of the selected sample, default: 'sample'
+        sampling_method : str
+            Method of sampling, 'topk' or 'lowk' or 'randk'
+        num_sample : int
+            Number of samples extracted
+        output_file : str
+            Path of sampler result, Use when user want to save results
+        """
+        super().__init__(extractor)
+
+        self.original_subset = subset_name
+        self.sampled_name = sampled_name
+        self.unsampled_name = unsampled_name
+
+        # optional. Use the --output_file option to save the sample list as a csv file.
+        if output_file is not None:
+            if output_file.split(".")[-1] != ".csv":
+                msg = f"Invalid extension, The extension of the file must end with .csv"
+                raise Exception(msg)
+
+        # 1. check the empty of the data set
+        if len(extractor) < 1:
+            msg = "Invalid Dataset, The dataset is empty."
+            raise Exception(msg)
+
+        # 2. Import data into a subset name and convert it
+        # to a format that will be used in the sampler algorithm with the inference result.
+        data_df, infer_df = self._load_inference_from_subset(extractor, subset_name)
+
+        # 3. Transfer the data to sampler algorithm to calculate uncertainty & get sample list
+        sampler = self._calculate_uncertainty(algorithm, data_df, infer_df)
+        self.result = sampler.get_sample(method=sampling_method, k=num_sample)
+
+        if output_file is not None:
+            self.result.to_csv(output_file, index=False)
+
+        self.sample_id = self.result["ImageID"].to_list()
+
+    @staticmethod
+    def _load_inference_from_subset(extractor, subset_name):
+        # 1. Get Dataset from subset name
+        if subset_name in extractor.subsets().keys():
+            subset = extractor.get_subset(subset_name)
+        else:
+            msg = f"Not Found subset '{subset_name}', available subsets: {extractor.subsets().keys()}"
+            raise Exception(msg)
+
+        data_df = defaultdict(list)
+        infer_df = defaultdict(list)
+
+        # 2. Fill the data_df and infer_df to fit the sampler algorithm input format.
+        for data in subset:
+            data_df["ImageID"].append(data.id)
+
+            if data.image is None:
+                msg = f"Invalid data, some data.image is None"
+                raise Exception(msg)
+            width, height = data.image.size
```

Comment:
`size` can return `None` in some cases (only `path` provided, but the image file is not available).